docs/source/writing-tests.md

```diff
@@ -304,7 +304,9 @@ There are several global variables you can use to introspect on Bats tests:
 - `$BATS_TEST_DESCRIPTION` is the description of the current test case.
 - `$BATS_TEST_NUMBER` is the (1-based) index of the current test case in the test file.
 - `$BATS_SUITE_TEST_NUMBER` is the (1-based) index of the current test case in the test suite (over all files).
-- `$BATS_TMPDIR` is the location to a directory that may be used to store temporary files.
+- `$BATS_TMPDIR` is the base temporary directory used by bats to create its
```

Comment:
As expounded in https://github.com/bats-core/bats-core/pull/365#issuecomment-778841245, I would propose to just mark `BATS_TMPDIR` as deprecated. Users should either go for `TMPDIR` or `BATS_RUN_TMPDIR`.

Reply:
> As expounded in #365 (comment), I would propose to just mark BATS_TMPDIR as deprecated. 

Indeed, but keep the logic as it is in the time being and mark it as **deprecated**

> Users should either go for TMPDIR or BATS_RUN_TMPDIR.

Only `TMPDIR`.
For me, even `BATS_RUN_TMPDIR` should not be documented since it is an `internal` variable automatically generated from `BATS_TMPDIR/TMPDIR` as prefix. (could be discussed on the other PR)

> I propose to enforce this decision with failing if BATS_TMPDIR is set when starting bats. Since BATS_TMPDIR was set to TMPDIR//tmp unconditionally, we should only "break" users who erroneously believed they were overriding it.

Starting with a warning could be a first step before throwing an error. WDTY ?

Reply:
I agree with keeping the logic and marking it as deprecated before removing it. Although it really does seem like `BATS_TMPDIR` is unconditionally set to `TMPDIR`/`/tmp`, there's a part of me that's leaning towards safe than sorry in this case.

Additionally, maybe we should move the internal variables outside of this list and perhaps under a new "Internal Variables" subsection? Or maybe we can at least mark which vars are internal and not meant for user config here in this list. (In the future, we really should have better conventions around internal variables - at least a naming/casing convention to differentiate between globals/externally-configured.)